### DISCUSSION TITLE
This implementation of OAuth2 not recommended for production environments.

### DISCUSSION THREAD

**Discussion by DDAK** [Kong/kong#12259] [Help] [answered]
Hi,

I want to use the OAuth2 on production. However, I am welcomed by this message on the documentation page:
```
Important:

This implementation of OAuth2 in Kong is for experimental or testing purposes only and is not recommended for production environments.
```
Any idea if this is ever going to be production ready?

Thanks.

ref: https://docs.konghq.com/hub/kong-inc/oauth2/

**Comment by StarlightIbuki** [comment, marked as answer]
We have the OpenID-Connect plugin for the enterprise version of Kong and it has the complete support for OAuth.
